plugins/Files/js/reducers/files.js

```diff
@@ -0,0 +1,71 @@
+import { Map, List } from 'immutable'
+import * as constants from '../constants/files.js'
+import { ls, searchFiles } from '../sagas/helpers.js'
+
+const initialState = Map({
+	activespending: '',
+	allocatedspending: '',
+	usage: 0,
+	files: List(),
+	workingDirectoryFiles: List(),
+	searchResults: List(),
+	uploading: List(),
+	downloading: List(),
+	path: '',
+	searchText: '',
+	uploadSource: '',
+	showAllowanceDialog: false,
+	showUploadDialog: false,
+	showSearchField: false,
+	showFileTransfers: false,
+	showDeleteDialog: false,
+	dragging: false,
```

Comment:
wowza that's a lotta state. We can't eliminate it -- this plugin has a lot of inherent state -- but we can break this apart into smaller sub-components. (Later.)